.dockerignore

```diff
@@ -1,2 +1 @@
 .idea/
-.output/
```

Comment:
why remove output? a lot of stuff goes in here, doesn't it?

Reply:
I removed it so that the docker context used for the [GKE](https://github.com/karlkfi/kpt-config-sync/blob/main/build/test-e2e-go/gke/Dockerfile) and [kind](https://github.com/karlkfi/kpt-config-sync/blob/main/build/test-e2e-go/kind/Dockerfile) e2e images could access the built config sync manifests when creating the image. This is only really an issue when running things inside these e2e containers (e.g. in CI).

I'm also fine with mounting the manifests directory to the container at runtime, although this seemed like a simpler approach to me.